### PR TITLE
commentor: Added commentor tool

### PR DESCRIPTION
**Pull request by D3vil0p3r**
Added PKGBUILD file for commentor tool (source: https://github.com/D3vil0p3r/comMENTOR)

### CODE REVIEW

**Review by noraj — COMMENTED** [review]
LGTM but I didn't tested

File: packages/commentor/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+pkgver() {
+  cd $pkgname
+  echo $(git rev-list --count HEAD).$(git rev-parse --short HEAD)
+}
+
+package() {
+  cd $pkgname
+  mkdir -p "$pkgdir/usr/share/commentor"
+  mkdir -p "$pkgdir/usr/bin"
+  install -Dm 755 -t "$pkgdir/usr/share/commentor/" commentor
```
Comment (noraj) [inline packages/commentor/PKGBUILD:26]: You could deploy multiple files in one line, cf our templates at https://github.com/BlackArch/blackarch-pkgbuilds
also

**Review by noraj — CHANGES_REQUESTED** [review]

File: packages/commentor/PKGBUILD (hunk 1 above)
Comment (D3vil0p3r) [reply packages/commentor/PKGBUILD]: Thank you @noraj for your tip. Solved on commit dd7124536ed647aee42aa7d99cad4bb919665036

**Comment by D3vil0p3r** [comment]
Hello @noraj Can you give a look to this PR?
Thanks

File: packages/commentor/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,29 @@
+pkgname=commentor
```
Comment (noraj) [inline packages/commentor/PKGBUILD:1]: you are missing the disclaimer comments https://github.com/BlackArch/blackarch-pkgbuilds/blob/66259a614509ab1f78075bc897d3cb10d832ed13/PKGBUILD-generic#L1-L2

File: packages/commentor/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+pkgver=19.d81a660
+pkgrel=1
+pkgdesc='Extract all comments from the specified URL resource.'
+groups=('blackarch' 'blackarch-code-audit')
```
Comment (noraj) [inline packages/commentor/PKGBUILD:5]: This don't really do any code audit but only extracts comments, I would rather in `blackarch-misc`

File: packages/commentor/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+arch=('any')
+url='https://github.com/D3vil0p3r/comMENTOR'
+license=('GPL3')
+depends=('curl' 'tidy' 'xmlstarlet')
+makedepends=('git')
+source=("git+https://github.com/D3vil0p3r/$pkgname.git")
+sha512sums=('SKIP')
+pkgver() {
```
Comment (noraj) [inline packages/commentor/PKGBUILD:13]: you need an empty line before `pkgver()`

File: packages/commentor/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  cd $pkgname
```
Comment (noraj) [inline packages/commentor/PKGBUILD:14]: add a blank line after cd

File: packages/commentor/PKGBUILD [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  echo $(git rev-list --count HEAD).$(git rev-parse --short HEAD)
+}
+
+package() {
```
Comment (noraj) [inline packages/commentor/PKGBUILD:18]: add `cd $pkgname`

File: packages/commentor/PKGBUILD [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  mkdir -p "${pkgdir}/usr/share/$pkgname"
```
Comment (noraj) [inline packages/commentor/PKGBUILD:19]: only use `$pkgdir` and not `${pkgdir}` when it's not needed, also use install and not mkdir eg. `install -dm 755 "$pkgdir/usr/bin"`

File: packages/commentor/PKGBUILD [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  mkdir -p "${pkgdir}/usr/bin"
+  install -Dm 755 -t "${pkgdir}/usr/share/$pkgname/" "${srcdir}/$pkgname/c_cleanup" "${srcdir}/$pkgname/c_decorate" "${srcdir}/$pkgname/c_filter" "${srcdir}/$pkgname/$pkgname"
```
Comment (noraj) [inline packages/commentor/PKGBUILD:21]: if you cd before you could drop all `${srcdir}/$pkgname`

**Review by noraj — CHANGES_REQUESTED** [review]

File: packages/commentor/PKGBUILD [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+package() {
+  cd $pkgname
+
+  install -dm 755 "$pkgdir/usr/share/$pkgname"
+  install -dm 755 "$pkgdir/usr/bin"
+  cd ${srcdir}/$pkgname
+  install -Dm 755 -t "$pkgdir/usr/share/$pkgname/" "c_cleanup" "c_decorate" "c_filter" "$pkgname"
+
+  cat > "${pkgdir}/usr/bin/$pkgname" << EOF
```
Comment (noraj) [inline packages/commentor/PKGBUILD:31]:
```suggestion
  cat > "$pkgdir/usr/bin/$pkgname" << EOF
```

**Review by noraj — CHANGES_REQUESTED** [review]

File: packages/commentor/PKGBUILD [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+  
+  echo $(git rev-list --count HEAD).$(git rev-parse --short HEAD)
+}
+
+package() {
+  cd $pkgname
+
+  install -dm 755 "$pkgdir/usr/share/$pkgname"
+  install -dm 755 "$pkgdir/usr/bin"
+  cd ${srcdir}/$pkgname
```
Comment (noraj) [inline packages/commentor/PKGBUILD:28]:
```suggestion
```

Comment (noraj) [reply packages/commentor/PKGBUILD:28]: This line is useless, this is the last thing to modify before merging.

**Review by noraj — CHANGES_REQUESTED** [review]

**Review by noraj — APPROVED** [review]
LGTM

**Comment by D3vil0p3r** [comment]
> LGTM

Among the reviewers above I don't see @noptrix that I think has the grants for merging the PR.

**Comment by noraj** [comment]
No need to add him as reviewer, he normally take a look quite often to the PR but those last weeks he has been overbusy so we may have to wait a little.

**Comment by noraj** [comment]
can you add `commentor` to lists/to-release too please?

**Comment by D3vil0p3r** [comment]
> can you add `commentor` to lists/to-release too please?

Added

**Pull request merged**